### PR TITLE
Preserve PDF annotations even when cache is reset

### PR DESCRIPTION
**Pull request by suhaibabsi-inst**
refs: [MBL-18069](https://instructure.atlassian.net/browse/MBL-18069)
affects: Student
release note: PDF annotations is preserved even when cache is reset

## Test Plan
* Login to Student app
* Open a PDF file and make some annotations to it, then go back.
* Close the app entirely.
* Go to the App's page on Settings app.
* Turn on "Reset cache on next launch" toggle.
* Launch the app, then login back to the same user. 
* Open the PDF file that was annotated in step 2.
* You should see the previously saved annotations on that file.

## Video Record

Recording covering PDF annotation persistence scenarios, which are: 
- On logout
- On cache reset.

https://github.com/user-attachments/assets/6c5e7355-bbff-4950-8fbc-982cb9d5f0fa

## Checklist

- [ ] Follow-up e2e test ticket created
- [ ] A11y checked
- [ ] Tested on phone
- [ ] Tested on tablet
- [ ] Tested in dark mode
- [ ] Tested in light mode
- [ ] Approve from product

[MBL-18069]: https://instructure.atlassian.net/browse/MBL-18069?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Student Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Build failed, skipping coverage check</td>
    </tr>
  </tbody>
</table>



#### Release Note: 
PDF annotations is preserved even when cache is reset

#### Affected Apps: Student

[MBL-18069](https://instructure.atlassian.net/browse/MBL-18069)

<details><summary>:x: XCTest failed: CoreTests/FileDetailsViewControllerTests/testMutatedPdfFileSavesToDocumentsDirectory</summary>


    XCTAssertTrue failed
</details>
<details><summary>:x: XCTest failed: CoreTests/FileDetailsViewControllerTests/testPrepLocalURLWithExistingPDFFile</summary>


    XCTAssertEqual failed: ("file:///Users/vagrant/Library/Developer/CoreSimulator/Devices/E9B79867-AB8C-4FE0-BC50-5206367C2D51/data/Containers/Data/Application/FA5E0EF0-2499-4F15-BF79-42944BCE64FF/tmp/canvas.instructure.com-1/1/File.pdf") is not equal to ("file:///Users/vagrant/Library/Developer/CoreSimulator/Devices/E9B79867-AB8C-4FE0-BC50-5206367C2D51/data/Containers/Data/Application/FA5E0EF0-2499-4F15-BF79-42944BCE64FF/Documents/canvas.instructure.com-1/1/File.pdf")
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against e5ea42ef0495e9d614e7f2e3d0f3ab59e9d6a180
</p>

File: Core/Core/Files/Model/LocalFileURLCreator.swift
Comment (vargaat) [inline Core/Core/Files/Model/LocalFileURLCreator.swift:38]: Modifying the load url like this will break access to already existing annotated files in the documents directory when the user updates the app to this version. Let's check for the annotated file on the original url as well OR create a job that moves all annotated pdfs to their new directory.

**Review by vargaat — CHANGES_REQUESTED** [review]

File: Core/Core/AppEnvironment/AppEnvironment.swift
Comment (suhaibabsi-inst) [inline Core/Core/AppEnvironment/AppEnvironment.swift:75]: I chose to call this here, and do it per session so to be targeting those folders relevant to this specific feature & session. Thus we do it lazily if there are multiple sessions.
There was an attempt to do it on app launch for all sessions at once, but there was a difficulty fetching all sessions that have files saved to `Documents` directory. Not all of them get persisted to Keychain as I was expecting.

**Review by ndrsszsz — DISMISSED** [review]
QA +1
Tested on iPhone 11, iOS 18.1

File: Core/Core/Login/LoginSession.swift
Comment (rh12) [inline Core/Core/Login/LoginSession.swift:164]: A `print` is left here

**Review by vargaat — CHANGES_REQUESTED** [review]

File: Core/Core/Login/LoginSession.swift
Comment (suhaibabsi-inst) [reply Core/Core/Login/LoginSession.swift]: Thanks for catching that! 👍

File: Core/Core/Login/LoginSession.swift
Comment (vargaat) [inline Core/Core/Login/LoginSession.swift:175]: Thanks for the filtering here. My concern with this is that in case we want to save something in the user's session folder in the future it will be copied to the annotated pdf directory since this only excludes the offline folder.
All the annotated pdfs are saved in directories with only numbers in their names and the annotated files have a pdf extension. Could we use this knowledge to move only such directories?

**Review by vargaat — CHANGES_REQUESTED** [review]

**Review by vargaat — APPROVED** [review]
QA+1, Thanks for the fixes!

**Review by vargaat — APPROVED** [review]

**Comment by suhaibabsi-inst** [comment]
This is still pending another review before merge.
cc: @ndrsszsz @rh12

**Review by ndrsszsz — APPROVED** [review]
QA +1

**Pull request merged**
